plerkle_messenger/src/redis/redis_messenger.rs

```diff
@@ -48,15 +42,84 @@ pub struct RedisMessenger {
 }
 
 pub struct RedisMessengerStream {
-    max_len: Option<StreamMaxlen>,
-    local_buffer: LinkedList<Vec<u8>>,
-    local_buffer_total: usize,
-    local_buffer_last_flush: Instant,
+    pub max_len: Option<StreamMaxlen>,
+    pub local_buffer: LinkedList<Vec<u8>>,
+    pub local_buffer_total: usize,
+    pub local_buffer_last_flush: Instant,
 }
 
-const REDIS_CON_STR: &str = "redis_connection_str";
-
 impl RedisMessenger {
+    pub async fn new(config: MessengerConfig) -> Result<Self, MessengerError> {
+        let uri = config
+            .get(REDIS_CON_STR)
+            .and_then(|u| u.clone().into_string())
+            .ok_or(MessengerError::ConfigurationError {
+                msg: format!("Connection String Missing: {}", REDIS_CON_STR),
+            })?;
+        // Setup Redis client.
+        let client = redis::Client::open(uri).unwrap();
+
+        // Get connection.
+        let connection = client.get_connection_manager().await.map_err(|e| {
```

Comment:
Can you say why its changed from `get_tokio_connection_manager`?

Reply:
It's deprecated in new Redis crate - https://docs.rs/redis/0.27.2/redis/struct.Client.html#method.get_tokio_connection_manager